source-firestore/pull.go

```diff
@@ -210,8 +258,10 @@ func (s *captureState) Validate() error {
 func (s *captureState) BindingIndex(resourcePath string) (int, bool) {
 	s.RLock()
 	defer s.RUnlock()
-	if state := s.Resources[resourcePath]; state != nil {
-		return state.bindingIndex, true
+	for _, state := range s.Resources {
```

Comment:
just sanity checking that this function (and friends) are not called super frequently ?

Reply:
Hmmm they actually are called very frequently, once per document in fact. I had been thinking of it in terms of not that many bindings to iterate through, but I guess that wouldn't necessarily be the case. I think we could hydrate a lookup table to use instead of looking through the entire list, so I'll do that.

Reply:
Alright I added a map to lookup the state key (for getting values from the state) from the resource path (which is what we get from captured documents). It should be more efficient if there are a lot of bindings. I think there's probably a more elegant way to do it, but I couldn't come up with anything that didn't seem like it would take a pretty deep reworking of things.